pliers/extractors/audio.py

```diff
@@ -488,4 +494,140 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
 
+
     _feature = 'percussive'
+
+
+YAMNET_INSTALL_MESSAGE = '''
+yamnet cannot be imported. To download and set up yamnet, open a terminal
+window and do the following
+- cd DOWNLOAD_PATH (path where you want yamnet to be downloaded)
+- git clone --depth 1 -b master https://github.com/tensorflow/models
+- cd models/research/audioset/yamnet
+- curl -O https://storage.googleapis.com/audioset/yamnet.h5
+- python yamnet_test.py
+
+If you're a Mac or Linux user:
+- open ~/.bash_profile
+- add "export PYTHONPATH=$PYTHONPATH:DOWNLOAD_PATH/models/research/audioset/yamnet" 
+to the end of the file
+- save and close
+
+If you're a Windows user, do:
+- set PYTHONPATH=%PYTHONPATH%;DOWNLOAD_PATH/models/research/audioset/yamnet
+- To make the change permanent, you have to add this line to your autoexec.bat
+'''
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+    
+    ''' Extract probability of 521 audio event classes based on AudioSet
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the audio segment (in seconds) on which label 
+        extraction is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If not defined, returns probabilities for all 521 labels.
+    label_subset (list): specifies subset of labels for which probabilities 
+        are to be returned. A comprehensive list of labels is available in the
+        audioset/yamnet repository (see yamnet_class_map.csv).
+    weights_path (optional): full path to model weights file. If not provided,
+        weights from pretrained YAMNet module are used.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'label_subset', 'weights_path',
+                       'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, label_subset=None,
+                 weights_path=None, **yamnet_kwargs):
+        try:
+            verify_dependencies(['yamnet'])
+        except MissingDependencyError: 
+            raise MissingDependencyError(dependencies=None,
+                                         custom_message=YAMNET_INSTALL_MESSAGE)
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+        self.weights_path = weights_path or path.join(MODULE_PATH, 'yamnet.h5')
+        self.hop_size = hop_size
+        self.yamnet_kwargs = yamnet_kwargs
+        self.params = yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+        for par, v in self.yamnet_kwargs.items():
+            if par in self.params.__dict__:
+                setattr(self.params, par, v)
+        self.labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
```

Comment:
maybe call this `self.all_labels` to differentiate from the labels selected in the end.

Reply:
decided to pass only a `self.labels`, which will be = `all_labels` if `label_subset` is not set, otherwise it is `label_subset` minus all non-existing labels passed by the user.

In `_extract`, the final set of labels returned in the df is stored in an extra `labels` variable, which will be either the same values of `self.labels` sorted by average probability across time points, or a subset of that (still sorted) if `top_n` is passed.